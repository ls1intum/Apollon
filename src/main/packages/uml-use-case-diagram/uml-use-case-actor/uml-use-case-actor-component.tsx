import React, { SFC } from 'react';
import { Text } from '../../../components/controls/text/text';
import { UMLUseCaseActor } from './uml-use-case-actor';

export const UMLUseCaseActorComponent: SFC<Props> = ({ element, scale }) => (
  <g>
    <rect width="100%" height="100%" fill="none" />
    <g stroke={element.strokeColor || 'black'} strokeWidth={2}>
      <circle cx={45 * scale} cy={25 * scale} r={15 * scale} />
      <line x1={45 * scale} y1={40 * scale} x2={45 * scale} y2={80 * scale} />
      <line x1={15 * scale} y1={55 * scale} x2={75 * scale} y2={55 * scale} />
      <line x1={45 * scale} y1={80 * scale} x2={15 * scale} y2={110 * scale} />
      <line x1={45 * scale} y1={80 * scale} x2={75 * scale} y2={110 * scale} />
    </g>
    <Text fill={element.textColor} x={45 * scale} y={130 * scale}>
      {element.name}
    </Text>
  </g>
);

interface Props {
  element: UMLUseCaseActor;
  scale: number;
}
