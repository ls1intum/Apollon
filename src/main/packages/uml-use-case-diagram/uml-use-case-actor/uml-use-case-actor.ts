import { DeepPartial } from 'redux';
import { UseCaseElementType } from '..';
import { ILayer } from '../../../services/layouter/layer';
import { ILayoutable } from '../../../services/layouter/layoutable';
import { IUMLElement, UMLElement } from '../../../services/uml-element/uml-element';
import { assign } from '../../../utils/fx/assign';
import { IBoundary } from '../../../utils/geometry/boundary';
import { UMLElementType } from '../../uml-element-type';

export class UMLUseCaseActor extends UMLElement {
  type: UMLElementType = UseCaseElementType.UseCaseActor;
  bounds: IBoundary = { ...this.bounds, width: 90, height: 140 };

  constructor(values?: DeepPartial<IUMLElement>) {
    super(values);
    assign<IUMLElement>(this, values);
  }

  render(canvas: ILayer): ILayoutable[] {
    return [this];
  }
}
