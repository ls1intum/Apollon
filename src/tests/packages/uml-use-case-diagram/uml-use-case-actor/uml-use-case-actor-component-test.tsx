import * as React from 'react';
import { render } from '@testing-library/react';
import { UMLUseCaseActor } from '../../../../main/packages/uml-use-case-diagram/uml-use-case-actor/uml-use-case-actor';
import { UMLUseCaseActorComponent } from '../../../../main/packages/uml-use-case-diagram/uml-use-case-actor/uml-use-case-actor-component';

it('render the uml-use-case-actor-component', () => {
  const actor: UMLUseCaseActor = new UMLUseCaseActor({ name: 'TestUseCaseComponent' });
  const { getByText, baseElement } = render(
    <svg>
      <UMLUseCaseActorComponent element={actor} scale={1.0} />
    </svg>,
  );
  expect(getByText(actor.name)).toBeInTheDocument();
  expect(baseElement).toMatchSnapshot();
});
