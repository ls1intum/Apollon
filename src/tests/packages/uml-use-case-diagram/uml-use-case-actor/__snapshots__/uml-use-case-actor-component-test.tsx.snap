// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the uml-use-case-actor-component 1`] = `
<body>
  <div>
    <svg>
      <g>
        <rect
          fill="none"
          height="100%"
          width="100%"
        />
        <g
          stroke="black"
          stroke-width="2"
        >
          <circle
            class="sc-jRQBWg cSVyHD"
            cx="45"
            cy="25"
            fill="white"
            r="15"
            stroke="black"
          />
          <line
            class="sc-furwcr TUFlT"
            stroke="black"
            x1="45"
            x2="45"
            y1="40"
            y2="80"
          />
          <line
            class="sc-furwcr TUFlT"
            stroke="black"
            x1="15"
            x2="75"
            y1="55"
            y2="55"
          />
          <line
            class="sc-furwcr TUFlT"
            stroke="black"
            x1="45"
            x2="15"
            y1="80"
            y2="110"
          />
          <line
            class="sc-furwcr TUFlT"
            stroke="black"
            x1="45"
            x2="75"
            y1="80"
            y2="110"
          />
        </g>
        <text
          dominant-baseline="middle"
          font-weight="bold"
          pointer-events="none"
          text-anchor="middle"
          x="45"
          y="130"
        >
          TestUseCaseComponent
        </text>
      </g>
    </svg>
  </div>
</body>
`;
